### PR TITLE
Fix #263 and minor build errors

### LINKED ISSUE #263: Nuke the compiler bridge cache when we publish the bridge
Copy from dotty:

```scala
      cleanSbtBridge := {
        val dottySbtBridgeVersion = version.value
        val dottyVersion = (version in `dotty-compiler`).value
        val classVersion = System.getProperty("java.class.version")

        val sbtV = sbtVersion.value
        val sbtOrg = "org.scala-sbt"
        val sbtScalaVersion = "2.10.6"

        val home = System.getProperty("user.home")
        val org = organization.value
        val artifact = moduleName.value

        IO.delete(file(home) / ".ivy2" / "cache" / sbtOrg / s"$org-$artifact-$dottySbtBridgeVersion-bin_${dottyVersion}__$classVersion")
        IO.delete(file(home) / ".sbt"  / "boot" / s"scala-$sbtScalaVersion" / sbtOrg / "sbt" / sbtV / s"$org-$artifact-$dottySbtBridgeVersion-bin_${dottyVersion}__$classVersion")
      },
      publishLocal := (publishLocal.dependsOn(cleanSbtBridge)).value,
```

https://github.com/lampepfl/dotty/blob/c84480960cf618c29705dbaab9332d304a081524/project/Build.scala#L534-L550

Idea from @smarter.

### PR DESCRIPTION
**Pull request by jvican**
This does three simple things:

* Updates the version.
* Nukes the compiler bridge caches.
* Formats the build sources.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -89,8 +89,17 @@ def altPublishSettings: Seq[Setting[_]] =
     }
   )
 
+val noPublish: Seq[Setting[_]] = List(
+  publish := {},
+  publishLocal := {},
+  publishArtifact in Compile := false,
```
Comment (dwijnand) [inline build.sbt:95]: You probably need to redefine `publishSigned` to fix #375.

**Review by dwijnand — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (eed3si9n) [reply build.sbt:95]: publishSigned respects `skip in publish` so that's going to be the fix I think.

File: build.sbt [hunk 2]
```diff
@@ -89,8 +89,17 @@ def altPublishSettings: Seq[Setting[_]] =
...
+  publishArtifact in Test := false,
+  publishArtifact := false,
+  skip in publish := true,
```
Comment (eed3si9n) [inline build.sbt:98]: 👍

**Review by eed3si9n — APPROVED** [review]

**Review by Duhemm — APPROVED** [review]
LGTM

File: build.sbt (hunk 1 above)
Comment (jvican) [reply build.sbt:95]: Can we actually remove the `publish*` and `publishArtifact` redefinitions if `skip in publish` is already being respected?

**Pull request merged**